### ISSUE TITLE
Adding dark reader to website using npm package not works when the extension is on

### ISSUE THREAD

**Issue by fainir** [darkreader/darkreader#2227]
I'm building a website and tried to implement dark reader inside of it. it works fine except when the dark reader extension is on. Can you tell me how can I fix it? (I use react)
Thanks a lot

**Comment by Gusted** [comment]
I assume you mean the npm package darkreader. So when you enable darkreader in the code and enable darkreader as extension. It doesn't display the site good.

**Comment by fainir** [comment]
Yes, sorry if I was unclear. Do you know what the reason is or what could be done to fix it?

**Comment by Gusted** [comment]
Well I cannot reproduce this easily but I think there's a bug in darkreader code. If you could provide your current site without any like critical information so I can reproduce this issue would help a lot.

**Comment by fainir** [comment]
I can't share the project that the bug accrued in because it isn't my code and I can't share it. I tried to reproduce it on other projects and the bug didn't accrue. So unfortunately I can't help you reproduce it :(
If you suspect that certain information about the project could lead you to resolve it, please ask me.
Sorry for not sharing the code and thank you for your help.

**Comment by Gusted** [comment]
No problem, will only a bit hard to understand why this problem could occur. Could you maybe try screenshot certain elements that aren't displayed correctly. And does the site use the normal darkreader.enabled() with the standard Filterconfig?

**Comment by fainir** [comment]
Yes, I understand the difficulty. The page doesn't load at all. it keeps loading the page forever 
![image](https://user-images.githubusercontent.com/6260146/78452502-31258000-7694-11ea-8159-978f519db468.png)

**Comment by fainir** [comment]
It doesn't show any errors in the log because it keep on loading the page.

**Comment by Gusted** [comment]
Could you remove darkreader in the code and try to load the page so it's confirmed to be darkreader sides

**Comment by Gusted** [comment]
@alexanderby bug label please

**Comment by fainir** [comment]
It works fine when the enableDark in the code without the extension. Also, it works fine when  the extension is on and I comment out the enableDark in the code

**Comment by Gusted** [comment]
It's like if the too will compete each other if it's enabled or not... Could you try to before making a darkreader call. To place a debugger; and than step for step see where in the code is the problem that wil help a lot

**Comment by fainir** [comment]
I did it and it didn't stop at the debugger it freeze the tab. Actually I didn't understand what do you mean by finding the problem in the code because the problem occurs only when enableDark function is added.

**Comment by fainir** [comment]
Another error I have but I dont think this is related to the bug is this error message:
![image](https://user-images.githubusercontent.com/6260146/78454904-b3697080-76a3-11ea-96fc-92d757be3e0d.png)

**Comment by Gusted** [comment]
Hmm that's very weird.... I will look into it. Also seems like this could be a simple cross origin bug that would just download that css and store it local host

**Comment by fainir** [comment]
adding  `setFetchMethod(window.fetch);` fixed the error in the console as was wirrten at the error.
Anyway, the problem with the endless loading still happening so if you could fix it somehow it could be wonderful :) I can provide you with more information if it could help.
Another approach is trying to detect weather the extension is on and if so to not run the code. Do you know if it could be done? because I couldn't find a solution for now.

**Comment by Gusted** [comment]
Okay you see my only logical solution is like what you said because I think it's now in the background toggling it every second because idk for what reason. But I think it's a good approach to see if it's already enabled

**Comment by n-peugnet** [comment]
I have the same problem on chromium and firefox. I am using the min version of the script from jsdeliver. You can check the problem with [this page](https://mers.club1.fr/film).

With the extension disbled both firefox and chromium renders the page without any issue.
![Screenshot from 2020-04-06 16-19-13](https://user-images.githubusercontent.com/23519418/78568559-91e2c300-7822-11ea-9ae0-8f65a37202c2.png)

With the extension disabled:
- On chromium the page loading never ends and there is only a dark background. After some time.
  ![Screenshot from 2020-04-06 16-20-14](https://user-images.githubusercontent.com/23519418/78568593-9c9d5800-7822-11ea-9479-f8136ef5d5ae.png)

- On firefox the page loading never ends and I get a strange render
  ![Screenshot from 2020-04-06 16-16-16](https://user-images.githubusercontent.com/23519418/78568621-a6bf5680-7822-11ea-82a5-f95fb3e3ec12.png)

**Comment by Gusted** [comment]
Okay this is very strange I cannot reproduce this.
https://i.imgur.com/sdKtvMb.gifv
I'm using Brave Version 1.7.83 Chromium: 80.0.3987.162 (Official Build) beta (64-bit)
I'm gonna try to reproduce this on Firefox.

**Comment by Gusted** [comment]
Cannot also reproduce this on Firefox 74.0 (64-bit)

This is very strange...

**Comment by n-peugnet** [comment]
I'm using firefox 74.0 and Chromium 80.0.3987.149 (Official Build) Built on Ubuntu, running on LinuxMint 19.3 (64-bit).

both with darkreader 4.9.4

**Comment by Gusted** [comment]
Do you any other Extension that maybe can interfer With Darkreader I'm running just darkreader only with Firefox

**Comment by n-peugnet** [comment]
I uninstalled every other extensions and made another test, but it still doesn't work.
![Peek 2020-04-06 23-19](https://user-images.githubusercontent.com/23519418/78606297-5dd8c380-785d-11ea-88f7-21d7a1fef29e.gif)

**Comment by Gusted** [comment]
This is very very Strange Gonna try to reproduce this one more time else IDK and need to bootup an old linux machine I guess

**Comment by n-peugnet** [comment]
I moved my test page and updated the link

**Comment by Gusted** [comment]
> adding `setFetchMethod(window.fetch);` fixed the error in the console as was wirrten at the error.
> Anyway, the problem with the endless loading still happening so if you could fix it somehow it could be wonderful :) I can provide you with more information if it could help.
> Another approach is trying to detect weather the extension is on and if so to not run the code. Do you know if it could be done? because I couldn't find a solution for now.

@fainir Take a look about this comment and issue. https://github.com/darkreader/darkreader/issues/1840#issuecomment-610308007
Please let me know if this helps your issue.

as of @n-peugnet Please make a seprate issue as it isn't the same origion

**Comment by fainir** [comment]
> > adding `setFetchMethod(window.fetch);` fixed the error in the console as was wirrten at the error.
> > Anyway, the problem with the endless loading still happening so if you could fix it somehow it could be wonderful :) I can provide you with more information if it could help.
> > Another approach is trying to detect weather the extension is on and if so to not run the code. Do you know if it could be done? because I couldn't find a solution for now.
> 
> @fainir Take a look about this comment and issue. [#1840 (comment)](https://github.com/darkreader/darkreader/issues/1840#issuecomment-610308007)
> Please let me know if this helps your issue.
> 
> as of @n-peugnet Please make a seprate issue as it isn't the same origion

the dark mode did not work when I added darkreader.js for some reason. but it did work when I used the cdn. I don't understand the reason because it works on the projects you sent.

**Comment by Gusted** [comment]
> > > adding `setFetchMethod(window.fetch);` fixed the error in the console as was wirrten at the error.
> > > Anyway, the problem with the endless loading still happening so if you could fix it somehow it could be wonderful :) I can provide you with more information if it could help.
> > > Another approach is trying to detect weather the extension is on and if so to not run the code. Do you know if it could be done? because I couldn't find a solution for now.
> > 
> > 
> > @fainir Take a look about this comment and issue. [#1840 (comment)](https://github.com/darkreader/darkreader/issues/1840#issuecomment-610308007)
> > Please let me know if this helps your issue.
> > as of @n-peugnet Please make a seprate issue as it isn't the same origion
> 
> the dark mode did not work when I added darkreader.js for some reason. but it did work when I used the cdn. I don't understand the reason because it works on the projects you sent.

yeah dakreader.js was temp and maybe the issue. You can use the CDN one as that have automaticly updates. 

So If I readt it right the issue has been resolved?

**Comment by fainir** [comment]
No, it didn't resolved. I meant that it became dark without the extension. but it didn't work when the extension is on as usual.

**Comment by Gusted** [comment]
Yeah the API is meant as it's own it doens't control the extension  
API != Extension
It's meant that the user disables darkreader when they use native dark theme 
or if it's by default and non toggle able add it into the dark sites list.

**Comment by fainir** [comment]
Yes I understand. the problem is that the website doesn't load when it my website uses the api and the extension is on. On other projects it works fine. Do you have any solution for this particular situation?

**Comment by Gusted** [comment]
Without the link to the site or an copy of the site source with ctrl + s or something (of course redacted personal and non viable stuff to the public) I don't really would know

**Comment by MoroseCorpse** [comment]
Hello, is there any news on this? Because I have the same problem

**Comment by Gusted** [comment]
I still cannot reproduce this.
Could you give some more info? @MoroseCorpse

**Comment by MoroseCorpse** [comment]
No, unfortunately I cannot tell you any more than the others already did. I have now anyway simply circumvented this with the following code:
```
if ($(".darkreader")[0]){
     console.log("Darkreader Extension detected");
} else {
     DarkReader.setFetchMethod(window.fetch);
     DarkReader.enable({
          brightness: 130,
          contrast: 110
     });
}
```
The only thing I can say is that I also had several mistakes like this: https://github.com/darkreader/darkreader/issues/2227#issuecomment-609046879

But I would like to ask you a question, even if it doesn't fit in here: Is it possible to set the API to invert or not invert certain elements?

**Comment by Gusted** [comment]
How do you mean not inverting it?

**Comment by MoroseCorpse** [comment]
I mean this function as in the Browser Extension, that you can set all elements for each web page individually by clicking on the point "extended" (German Version - I don't know if it says so in the English version)

**Comment by Gusted** [comment]
So you actually asking:

`Einige Elemente auf der Website möchte ich nicht von Darkreader angefasst werden`
I speak a little bit German and with a good translator 😅

**Comment by MoroseCorpse** [comment]
Let me try to explain it another way:
English (Translated by Deepl):
I would like to have the settings, which you also have in the browser extension
German:
Ich möchte gerne die Einstellungen haben, die man in der Browser Extension auch hat

![jwfz5yL](https://user-images.githubusercontent.com/44735715/80421167-55aff900-88dc-11ea-9059-3894139e5034.png)
![tO0XOYm](https://user-images.githubusercontent.com/44735715/80421174-58125300-88dc-11ea-933a-626099183de5.png)

Something like this, but for the Web API

**Comment by Gusted** [comment]
Pfff to make fixes to your own site I think, I can look into it to add an function in the API

**Comment by MoroseCorpse** [comment]
Yeah that would be really nice if you could add such a function

**Comment by Gusted** [comment]
I will look into it.

**Comment by MoroseCorpse** [comment]
This is not to sound bad, but how long do you think it could take to complete the function?

**Comment by Gusted** [comment]
> This is not to sound bad, but how long do you think it could take to complete the function?

This week? IDK I've a lot of things to code for darkreader V5 and etc, but as I'm doing a function that require some changes in the API module it won't be long

**Comment by MoroseCorpse** [comment]
Okay, that sounds good. Thanks in advance

**Comment by Gusted** [comment]
Other things came up, I'm sorry. Dark Reader V5 is a big thing. 

But I managed to rather some time and made a good Fix for this #2855
